src/main/java/org/elasticsearch/index/snapshots/blobstore/SnapshotFiles.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.elasticsearch.index.snapshots.blobstore;
+
+import com.google.common.collect.ImmutableList;
+import org.elasticsearch.index.snapshots.blobstore.BlobStoreIndexShardSnapshot.FileInfo;
+/**
+ */
+public class SnapshotFiles {
+
+    private final String snapshot;
+
+    private final ImmutableList<FileInfo> indexFiles;
+
+    public String snapshot() {
+        return snapshot;
+    }
+
+    public SnapshotFiles(String snapshot, ImmutableList<FileInfo> indexFiles ) {
```

Comment:
Would it be worth transforming the list of `FileInfo`s into a map when `SnapshotFiles` is created so that it doesn't have to look through the list every time for `containPhysicalIndexFile` and `findPhysicalIndexFile`?